### ISSUE TITLE
Fix/implement getMoreLikeThisIdString()

### ISSUE THREAD

**Issue by irnnr** [TYPO3-Solr/ext-solrmlt#4]
Need to implement the plugin's getMoreLikeThisIdString() method in a generic way. Right now the function is simply broken. 

Historical background: EXT:solrmlt was developed for a showcase on www.typo3-solr.com where we indexed the TYPO3 mailing list archives through TYPO3's newsgroup mirror (NNTP). When viewing a search result which is a mailing list message / newsgroup message we would list similar messages in a side bar. These messages were picked by providing EXT:solrmlt with a GET parameter tx_solrnntp[id]. A MLT query would then be created using that ID. Currently the old NNTP related code is simply commented out and replaced by the current page's title.
The title however is not an ID. 

https://github.com/TYPO3-Solr/ext-solrmlt/blob/master/Classes/Plugin/MoreLikeThis.php#L120

**Comment by irnnr** [comment]
looking at how it is used ATM, the method could be renamed `getMoreLikeThisQueryString()` as well.

**Comment by timohund** [comment]
I've refactored the handling of the plugin in a branch and moved the configuration to a configuration object and the creation of the query to a query builder, which works quite well.

The query is now retrieved from the page title or could be build from the page id:

https://github.com/timoschmidt/ext-solrmlt/commit/420282dac6c244622e611b74a57f82ca1a6dafe6#diff-cc181afcea2954aae679ca722a436b64R75

Currently the plugin behaviour is:
for the type "pagetitle": Everything is working as before
for the type "pageid": I only get the document itself back, nothing else that could be related.

Any idea? I would say in any case it makes sence to keep the refactoring. How should we go on with quering by "id:..."?

**Comment by timohund** [comment]
@irnnr What do you think?

**Comment by irnnr** [comment]
I think I found it; this describes the behavior for using content (at the bottom)
https://wiki.apache.org/solr/MoreLikeThisHandler

So we'd have to get the page's content and then POST it to MLT. We could simply get the page's content by querying Solr I believe. Otherwise we'd have to "crawl" the page and extract content like in EXT:solr. I would not want to do that.

Alternatively/In addition to the page title and the page content I think it might make sense to offer other fields as well, f.e. meta keywords, description and so on. Probably needs some testing how well that works.

**Comment by irnnr** [comment]
Here's more information in general: https://cwiki.apache.org/confluence/display/solr/MoreLikeThis

**Comment by dkd-dobberkau** [comment]
hi. i wonder if MLT would also work on just the document id that the current context has. I would assume that we only can show indexed documents and their resembling documents. 
Addtionally MLT should be loaded via an e-id iframe or ajax call.

**Comment by irnnr** [comment]
I'll have to look at MLT in detail again I guess. 

I think the current "static" way should be good for now, otherwise it would further delay the release. Maybe we can add it for the next release. I think it does make sense though, especially if we could make it work so that we could use it for things like ecommerce/related products.

**Comment by timohund** [comment]
@irnnr 
@dkd-dobberkau 

Hi together,

i had a look on this today again and i think this is bigger then expected.
1. The documentation is about the MoreLikeThisHandler using q=id:xx is only supported when you are using the this MoreLikeThisHandler! Currently solrmlt is only using the normal /select Handler, which is only offering some parts of "MoreLikeThis" (By using the MoreLikeThisComponent).

The consenquence should be:

a) We need to enable the mlt QueryHandler "/mlt" in EXT:solr. Currently this handler is only configured on "mlt" (without the leading /) which makes no sence for me. In addition the SolrService needs to support to query the Endpoint "/mlt".

So before we can extend solrmlt we need to prepare this in EXT:solr.

The pull request as it is right now (https://github.com/TYPO3-Solr/ext-solrmlt/pull/8) provides the same behaviour as the extension had before (Find similar documents by the title of the page where the document is included). 

I think we should:
- Merge this pull request because it refactores the structure and also adds Tests and Travis integration
- Maybe we can do a release after that because the extension is working in general as before and compatible to EXT:solr 3.1.1
- We should prepare the usage of "/mlt" in EXT:solr
- Afterwards we could extens solrmlt to really use the "/mlt" Handler.

What do you think?

**Comment by timohund** [comment]
The renaming and refactoring was done. For querying by "id" an new Issue will be created for release milestone 1.2
